docs/rackhd/graphs.rst

```diff
@@ -31,6 +31,287 @@ Graphs are defined via a JSON definition that conform to this schema:
+            {
+                "label": "task-1",
+                "injectableName": "Task.example"
+            },
+            {
+                "label": "task-2",
+                "injectableName": "Task.example",
+                "waitOn": {
+                    "task-1": "succeeded"
+                }
+            },
+            {
+                "label": "task-3",
+                "injectableName": "Task.example",
+                "waitOn": {
+                    "task-1": "succeeded"
+                }
+            }
+        ]
+    }
+
+This graph is almost the same as the "Linear ordered tasks" example, except that ``task-2`` and ``task-3`` *both* have a dependency on ``task-1``. When
+``task-1`` succeeds, ``task-2`` and ``task-3`` will be started in parallel.
+
+Tasks can also be ordered based on multiple dependencies:
+
+::
+
+    {
+        "injectableName": "Graph.Example.MultipleDependencies",
+        "friendlyName": "Tasks with multiple dependencies",
+        "tasks": [
+            {
+                "label": "task-1",
+                "injectableName": "Task.example"
+            },
+            {
+                "label": "task-2",
+                "injectableName": "Task.example"
+            },
+            {
+                "label": "task-3",
+                "injectableName": "Task.example",
+                "waitOn": {
+                    "task-1": "succeeded",
+                    "task-2": "succeeded"
+                }
+            }
+        ]
+    }
+
+In the above example, ``task-1`` and ``task-2`` will be started in parallel, and ``task-3`` will only be started once ``task-1`` and ``task-2`` have both succeeded.
+
+.. _Graph Options:
+
+**Graph Options**
+
+As detailed in the :ref:`task-definition-ref-label` section, each task definition has an options object 
+that can be used to customize the task. All values set in the options objects are considered defaults, and can be overridden
+within the Graph definition. Additionally, the options values can be overridden again by the data in the API request made to run the graph.
+
+For exampe, a simple task definition with options looks like this:
+
+::
+
+    {
+        "injectableName": "Task.Example.Options",
+        "friendlyName": "Task with basic options",
+        "implementsTask": "Task.Base.Example",
+        "options": {
+            "option1": "value 1",
+            "option2": "value 2"
+        },
+        "properties": {}
+    }
+
+As is, this task definition specifies default values of "value 1" and "value 2" for its respective options. 
+In the graph definition, these values can be changed to have new defaults by adding a key to the ``Graph.options`` object
+that matches the ``label`` string given to the task object ("example-options-task" in this case):
+
+::
+
+    {
+        "injectableName": "Graph.Example.Options",
+        "friendlyName": "Override options for a task",
+        "options": {
+            "example-options-task": {
```

Comment:
^^ is the name `example-options-task` specific here? Can you have more than one? If they specify the same keys within two different sets, what happens? e.g.

```
"options": {
    "foo": {
        "option1": "foo overridden option"
    },
    "bar": {
        "option1": "bar overridden option"
    },
```
 what would the tasks run with?